### PR TITLE
Animated number counter

### LINKED ISSUE #36: Some UI Suggestions
- UI of this section can be improved by using some type of animated stats counter, [see here](https://elfsight.com/number-counter-widget/) (see the below section in the above link)

- Lower section of homepage looks a bit clutterd, may be little bit of spacing between components can improve UI

### PR DESCRIPTION
**Pull request by ayaankhan98**
closes #36

### CODE REVIEW

File: components/Counter.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-      </div>
+      <div className="flex mx-2 my-5 lg:my-40 md:my-14 lg:mx-20 justify-around lg:text-4xl md:text-2xl text-blue-850 font-bold" style={{fontSize: "80px", display: "flex"}} >
+        <div className="flex-shrink">
+          <CountUp start={0} end={20} duration={1} />+ <span style={{fontSize: "40px"}}>Countries</span>
+        </div>
+        <div className="flex-shrink">
+          <CountUp start={0} end={99} duration={1} />+ <span style={{fontSize: "40px"}}>Universities</span>
+        </div>
+        <div className="flex-shrink">
+          <CountUp start={0} end={119} duration={1} />+ <span style={{fontSize: "40px"}}>Scholars</span>
```
Comment (ayaankhan98) [inline components/Counter.js:13]: Just add a `+` sign there so decreased 22 to 20, 100 to 99 and 120 to 199

**Review by roshnet — CHANGES_REQUESTED** [review]
@ayaankhan98 thanks for the changes. It looks good on desktop, however doesn't fit on mobile devices. Please check screenshot -

image

**Comment by ayaankhan98** [comment]
I think this looks fine? isn't?
Screenshot_20210829_141753

**Comment by ayaankhan98** [comment]
A better way to do this is show count animation only when Counter component comes into picture (meaning in viewport). The way it's working right now is it starts as soon as the web page loads, and may be it will not make any sense because it will give static view as earlier if the user does not scrolls to that component in mobile view on load or in desktop mode (if component is out of viewport onload).

A simple solution to this is to create a custom hook to check the visibility of a component in view port or may be track the window scroll position and check if your div comes into picture or not (Perhaps are other better ways to achieve the same).

May be someone else can improve based on this.

**Comment by roshnet** [comment]
> @roshnet 
> I think this looks fine? isn't?
> Screenshot_20210829_141753
> 

@ayaankhan98 yes, this is perfect.

**Comment by roshnet** [comment]
> A better way to do this is show count animation only when Counter component comes into picture (meaning in viewport). The way it's working right now is it starts as soon as the web page loads, and may be it will not make any sense because it will give static view as earlier if the user does not scrolls to that component in mobile view on load or in desktop mode (if component is out of viewport onload).
> 
> A simple solution to this is to create a custom hook to check the visibility of a component in view port or may be track the window scroll position and check if your div comes into picture or not (Perhaps are other better ways to achieve the same).
> 
> May be someone else can improve based on this.

Good idea! Although since I think the current PR is good enough, this enhancement can be implemented in another issue.

That said, feel free to go ahead an update the PR if you've got an idea, thanks!

**Comment by ayaankhan98** [comment]
> Good idea! Although since I think the current PR is good enough, this enhancement can be implemented in another issue.

I think this is good, may be someone else will do this.

**Comment by himanshuvarandani** [comment]
@ayaankhan98 changes looks good, but this doesn't fit for tablet view and large screens (768px < size < 1200px).
Check screenshots:

image

image


I can suggest some changes: 
-  decrease the size for sm and md screens 
-  put all 3 counters in the same line and write all counter label in below line (check the second screenshot you gave in the issue #36)

**Comment by ayaankhan98** [comment]
> @ayaankhan98 changes looks good, but this doesn't fit for tablet view and large screens (768px < size < 1200px).

Hope now this looks good over all commonly used resolutions.

**Comment by pnijhara** [comment]
@roshnet if it is ready to merge then you can merge it. You can also open the new issue related to scrolling that Ayan pointed out.

**Pull request merged**